File: backend/app/application/dto/care_log_dto.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from app.domain.entities.care_log import CareCategory


@dataclass
class CreateCareLogDTO:
    baby_id: UUID
    category: CareCategory
    recorded_at: datetime
    name: str | None = None
    dose: str | None = None
    memo: str | None = None
    id: UUID | None = None  # 클라 생성 UUID(멱등 upsert)


@dataclass
class UpdateCareLogDTO:
    id: UUID
    category: CareCategory
    recorded_at: datetime
    name: str | None = None
    dose: str | None = None
    memo: str | None = None


@dataclass
class CareLogResponseDTO:
    id: UUID
    baby_id: UUID
    category: CareCategory
    name: str | None
    dose: str | None
    recorded_at: datetime
    memo: str | None
    created_at: datetime

File: zzippu_ios/zzippu/Data/Network/DTOs/CareLogDTO.swift
// Data/Network/DTOs/CareLogDTO.swift
// 서버 CareLogResponse / CareLogCreateRequest / CareLogUpdateRequest 대응
// snake_case ↔ camelCase 는 APIClient 의 convert 전략이 처리(recordedAt↔recorded_at 등).

import Foundation

// MARK: - Response DTO

struct CareLogResponseDTO: Decodable {
    let id: UUID
    let babyId: UUID
    let category: String
    let name: String?
    let dose: String?
    let recordedAt: Date
    let memo: String?
    let createdAt: Date
}

// MARK: - Create Request DTO

struct CareLogCreateRequestDTO: Encodable {
    let category: String
    let name: String?
    let dose: String?
    let recordedAt: Date
    let memo: String?
}

// MARK: - Update Request DTO

struct CareLogUpdateRequestDTO: Encodable {
    let category: String        // 백엔드 필수(카테고리 불변이라도 전송) — 누락 시 422
    let name: String?
    let dose: String?
    let recordedAt: Date        // 백엔드 필수 — 옵셔널 아님
    let memo: String?
}
